src/status_im/anon_metrics/core.cljs

```diff
@@ -0,0 +1,94 @@
+(ns status-im.anon-metrics.core
+  (:require [status-im.ethereum.json-rpc :as json-rpc]
+            [taoensso.timbre :as log]
+            [re-frame.core :as re-frame]
+            [re-frame.interceptor :refer [->interceptor]]
+            [status-im.utils.async :refer [async-periodic-exec async-periodic-stop!]]
+            [status-im.utils.platform :as platform]
+            [status-im.utils.build :as build]
+            [status-im.utils.fx :as fx]
+            [status-im.anon-metrics.transformers :as txf]))
+
+(defonce events-foyer (atom []))
+(defonce periodic-tasks-chan (atom nil))
+
+(defn onboard-events
+  "Check if there are any events in the foyer, flush them to the backend and clear foyer on-success."
+  []
+  (let [outstanding-events @events-foyer]
+    (when (seq outstanding-events)
+      (reset! events-foyer [])
+      (json-rpc/call {:method     "appmetrics_saveAppMetrics"
+                      :params     [outstanding-events]
+                      :on-error   (fn [err]
+                                    (log/error {:error err
+                                                :events outstanding-events})
+                                    (log/warn "All outstanding events will be rejected"))}))))
+
+(re-frame/reg-fx
+ ::transfer-data
+ (fn [transfer?]
+   (if (and transfer?
+            ;; double run safety
+            (not @periodic-tasks-chan))
```

Comment:
i mean here, if you stop it and then try to run again it won't work, because we didn't reset it to nil

Reply:
Sorry, that's so stupid of me. I'll fix it. Thank you.